File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int count=0;
        for(int i=1;i<arr.size()-1;i++){
            if(arr[i]>arr[i-1] && arr[i]>arr[i+1]){
                count = i;
                break;
            }
                
        }
        return count;
    }
};

File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int low = 0;
        int high = arr.length-1;
        
        while(low<high){
            int mid = low + (high-low)/2;
            
            if(arr[mid] < arr[mid+1]){
                low = mid+1;
            }
            else {
                high = mid;
            }
        }
        return high;
    }
}

/*
//Approach 1 - BruteForce o(n)
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        for(int i=1;i<arr.length-1;i++){
            if(arr[i]>arr[i-1] && arr[i]>arr[i+1])
                return i;
        }
        return 1;
    }
}
*/